### PR TITLE
Fix windows cross compilation

### PR DESCRIPTION
**Pull request by Jake-Shadle**
# Description
When compiling to windows from any host other than windows, the `win-exception-handler` crate would not actually compile the C code since the compilation was guarded by `#[cfg(target_os = "windows")]`, which is evaluated when the build script itself is compiled which would mean it would != windows, which would eventually cause a link error due to the missing library.

This change gets the target os via the `CARGO_CFG_TARGET_OS` environment variable, which is set by cargo when invoking the build script, and will evaluate to `windows` when actually targetting windows, no matter the host platform.

It also changed from using cmake to use the cc crate to compile the code, as cmake seemed like massive overkill for a single C file, as well as complicating cross compilation again, though I can change it back to cmake if you don't want that change.

I also fixed a warning produced by clang by adding `#include <intrin.h>`, though there was another warning due to the trampoline function pointer taking a non-const `wasmer_instance_context_t*`, but wasmer actually sending in a const *.

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

File: lib/win-exception-handler/build.rs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 fn main() {
-    #[cfg(target_os = "windows")]
-    {
-        use cmake::Config;
-        let project_name = "exception_handling";
-        let dst = Config::new(project_name).build();
-        println!("cargo:rustc-link-search=native={}", dst.display());
-        println!("cargo:rustc-link-lib=static={}", project_name);
```
Comment (MarkMcCaskey) [inline lib/win-exception-handler/build.rs]: Why are these lines deleted?  Surely, we still need to link the compiled output, or does `cc` somehow do that as well?

Comment (Jake-Shadle) [reply lib/win-exception-handler/build.rs]: Sorry, yes, cc automatically

1. Determines the compiler to use based on the environment
2. Specifies additional linker search directories
3. Additional libraries to link to
